tests/lib/thrift/ThriftTest/src/TestUtils.hpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2022 Young Mei
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <cstdio>
+#include <iostream>
+#include <sstream>
+
+#include <ztest.h>
+
+#define zassert_exception(msg, ...) zassert(0, "Caught unhandled exception: ", msg, ##__VA_ARGS__)
+
+#define BASETYPE_IDENTITY_TEST(func, instance, value)                                              \
```

Comment:
Let's avoid doing this please and just use `zassert_*()` inside of test functions. The test should already fail if an exception is thrown, so there is no need to make an assertion.